WinUIB2CApp/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,248 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MSALClientHelper"/> class.
+        /// </summary>
+        public MSALClientHelper(AzureADB2CConfig azureADB2CConfig)
+        {
+            this.AzureADB2CConfig = azureADB2CConfig;
+
+            this.InitializePublicClientApplicationBuilder();
+        }
+
+        /// <summary>
+        /// Initializes the MSAL's PublicClientApplication builder.
+        /// </summary>
+        /// <autogeneratedoc />
+        private void InitializePublicClientApplicationBuilder()
+        {
+            this.PublicClientApplicationBuilder = PublicClientApplicationBuilder.Create(AzureADB2CConfig.ClientId)
+                .WithExperimentalFeatures() // this is for upcoming logger
+                .WithB2CAuthority($"{AzureADB2CConfig.Instance}/tfp/{AzureADB2CConfig.Domain}/{AzureADB2CConfig.SignUpSignInPolicyid}")
+                .WithRedirectUri($"msal{AzureADB2CConfig.ClientId}://auth")
+                .WithLogging(new IdentityLogger(EventLogLevel.Warning), enablePiiLogging: false)    // This is the currently recommended way to log MSAL message. For more info refer to https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/logging. Set Identity Logging level to Warning which is a middle ground
+                .WithIosKeychainSecurityGroup("com.microsoft.adalcache");
+        }
+
+        /// <summary>
+        /// Initializes the public client application of MSAL.NET with the required information to correctly authenticate the user.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<IAccount> InitializePublicClientAppAsync()
+        {
+            // Initialize the MSAL library by building a public client application
+            this.PublicClientApplication = this.PublicClientApplicationBuilder.Build();
+
+            await AttachTokenCache();
+            return await FetchSignedInUserFromCache().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Initializes the public client application of MSAL.NET with the required information to correctly authenticate the user.
+        /// </summary>
+        /// <returns>An IAccount of an already signed-in user (if available)</returns>
+        public async Task<IAccount> InitializePublicClientB2CAppAsync(IntPtr? handle)
+        {
+            // Initialize the MSAL library by building a public client application for authenticating using WAM
+            this.PublicClientApplication = this.PublicClientApplicationBuilder
+                    .Build();
+
+            await AttachTokenCache();
+            return await FetchSignedInUserFromCache().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Attaches the token cache to the Public Client app.
+        /// </summary>
+        /// <returns>IAccount list of already signed-in users (if available)</returns>
+        private async Task<IEnumerable<IAccount>> AttachTokenCache()
```

Comment:
Naming: all async methods should end in `Async`

AttachToCache should not return anything.